### ISSUE TITLE
question on --update-freq and warmup-updates with transformer

### ISSUE THREAD

**Issue by robertodessi** [pytorch/fairseq#263]
When training a transformer model with --update-freq, do you also change --warmup-updates to a lower number?

I am currently training a base transformer model and due to memory constrains I am using a batch size of 16 and --update freq. The training set is around 110k. 

EDIT: I forgot to say I am using adam with the same configuration reported in the README of the examples

```
  --optimizer adam --adam-betas '(0.9, 0.98)' --clip-norm 0.0 \
  --lr-scheduler inverse_sqrt --warmup-init-lr 1e-07 --warmup-updates 4000 \
  --lr 0.0005 --min-lr 1e-09 \
```

But I am planning to increase the lr in the next experiment given the simulated larger batch size

**Comment by myleott** [comment]
I tried experimenting with the number of warmup updates in the past. In some cases you can get faster training with a smaller warmup, but it was brittle so I usually leave it at 4k even when training with big batches.

**Comment by robertodessi** [comment]
Thanks for your reply! I'll experiment with warmup updates with this in mind then.
